src/libraries/System.Reflection.Emit/tests/ModuleBuilder/ModuleBuilderDefineEnum.cs

```diff
@@ -160,11 +160,11 @@ public void DefineEnum_VoidUnderlyingType_ThrowsArgumentException()
         }
 
         [Fact]
-        public void DefineEnum_ByRefUnderlyingType_ThrowsCOMExceptionOnCreation()
+        public void DefineEnum_ByRefUnderlyingType_ThrowsTypeLoadExceptionOnCreation()
         {
             ModuleBuilder module = Helpers.DynamicModule();
             EnumBuilder enumBuilder = module.DefineEnum("Name", TypeAttributes.Public, typeof(int).MakeByRefType());
-            Assert.Throws<COMException>(() => enumBuilder.CreateTypeInfo());
+            Assert.Throws<TypeLoadException>(() => enumBuilder.CreateTypeInfo());
```

Comment:
This may be fixing https://github.com/dotnet/runtime/issues/43282 . This issue also mentions unnessary IsByRef check in GetTypeTokenWorkerNoLock : https://github.com/dotnet/runtime/issues/43282#issuecomment-726895475 . You may want to delete it too.

Reply:
Agree.

@steveharter Can you elaborate on the tests needed for removing https://github.com/dotnet/runtime/blob/6f8d8b20a0c78dcfc09ea7a43db48a9a4e055e13/src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/ModuleBuilder.cs#L1088-L1091

Reply:
This standalone program that hits this path: 
```
using System.Reflection;
using System.Reflection.Emit;

var ab = AssemblyBuilder.DefineDynamicAssembly(new AssemblyName("MyAssembly"), AssemblyBuilderAccess.Run);
var modb = ab.DefineDynamicModule("MyModule");
var tb = modb.DefineType("MyType", TypeAttributes.Public);
var mb = tb.DefineMethod("MyMethod", MethodAttributes.Public | MethodAttributes.Static, typeof(Type), Type.EmptyTypes);
var ilg = mb.GetILGenerator();

ilg.Emit(OpCodes.Ldtoken, typeof(int).MakeByRefType()); 
ilg.Emit(OpCodes.Ret);
```
To turn it into a test, refactor it to follow similar pattern as `DefineByRefField` tests added in this PR.

Reply:
> @steveharter Can you elaborate on the tests needed for removing https://github.com/dotnet/runtime/blob/6f8d8b20a0c78dcfc09ea7a43db48a9a4e055e13/src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/ModuleBuilder.cs#L1088-L1091

These are the tests that verify by-ref-type is not allowed:
[TypeBuilderDefineEvent.DefineEvent_ByRefEventType_ThrowsArgumentException](https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/libraries/System.Reflection.Emit/tests/TypeBuilder/TypeBuilderDefineEvent.cs#L108-L114)

[TypeBuilderAddInterfaceImplementation.AddInterfaceImplementation_ByRefInterfaceType_ThrowsArgumentExceptioN](https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/libraries/System.Reflection.Emit/tests/TypeBuilder/TypeBuilderAddInterfaceImplementaion.cs#L64-L69)

[TypeBuilderSetParent.SetParent_ByRefType_ThrowsArgumentExceptionOnCreation](https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/libraries/System.Reflection.Emit/tests/TypeBuilder/TypeBuilderSetParent.cs#L58-L67)

[TypeBuilderDefineNestedType.DefineNestedType_ByRefInterfaceType_ThrowsArgumentException](https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/libraries/System.Reflection.Emit/tests/TypeBuilder/TypeBuilderDefineNestedType.cs#L213-L218)